src/makeInfinityAdder.js

```diff
@@ -4,7 +4,22 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+    sum += num;
+
+    return adder;
```

Comment:
The `adder` function does not check if the `num` argument is a number. Although the task description states that there is no need to check if the argument is a number, this could lead to unexpected behavior if non-numeric values are passed. Consider adding a check to ensure `num` is a number to prevent potential issues.